buildbot/buildbot/worker.py

```diff
@@ -360,11 +361,18 @@ def stage_fpga_execute(db, config):
             task.log('skipping FPGA execution stage')
             return
 
+        # Copy datafiles to executable dir.
+        data_files = [os.path.join(task.code_dir, f) for f in os.listdir(task.code_dir) if f.endswith('.data')]
+        dest = os.path.join(task.code_dir, 'sd_card')
+        task.run(
+            ['cp'] + data_files + [dest]
+        )
```

Comment:
It seems like it would be a good idea to skip this if `data_files` is empty (put it under `if data_files:`). Perhaps a safer way to do this would be to use [shutil.copy](https://docs.python.org/3.7/library/shutil.html#shutil.copy) instead of `cp`.

Reply:
condition check is included now. I didn't use shutil.copy yet.

Reply:
It seems shutil.copy permits copying files one at a time. Unless there's an additional benefit, I'll mark the PR ready.